### ISSUE TITLE
Documentation of m4 substitution variables with optional library support

### ISSUE THREAD

**Issue by Julien-Elie** [rra/rra-c-util#10]
The comment is not right in three m4 macros (_bdb.m4_, _sasl.m4_, _zlib.m4_).  For instance:

```
dnl Provides the RRA_LIB_BDB_OPTIONAL macro, which should be used if Berkeley
dnl DB support is optional.  This macro will still always set the substitution
dnl variables, but they'll be empty unless --with-bdb is given.
```

It should rather be `they'll be empty unless Berkeley DB libraries are found and --without-bdb is not given`.

The comment in _libevent.m4_, _openssl.m4_, _pcre.m4_, _remctl.m4_ and `sqlite3.m4` could be homogenized in a similar way.  It currently merely says `If it isn't found, the substitution variables will be empty` or a variation with `unless the library is found` without mentioning the `--without-xxx` flag.

Also, there's a typo in "ecurrent" in _kadm5clnt.m4_.

**Comment by Julien-Elie** [comment]
Thanks Russ.  I think there is a typo in _kadm5clnt.m4_: `sets rra_use_KADM5CLNT to true if the SQLite library is found` should hint at the kadmin client instead.

**Comment by rra** [comment]
Thanks, copy/paste error. Fixed in c4b92649af268dc138bd24a001882b261c177537.
